src/dataframe/dataframe.jl

```diff
@@ -182,24 +227,36 @@ function DataFrame(pairs::Pair{<:AbstractString,<:Any}...; makeunique::Bool=fals
                      copycols=copycols)
 end
 
-# these two are needed as a workaround Tables.jl dispatch
-DataFrame(pairs::AbstractVector{<:Pair}; makeunique::Bool=false,
-          copycols::Bool=true) =
-    DataFrame(pairs..., makeunique=makeunique, copycols=copycols)
-
-DataFrame(pairs::NTuple{N, Pair}; makeunique::Bool=false,
-          copycols::Bool=true) where {N} =
-    DataFrame(pairs..., makeunique=makeunique, copycols=copycols)
+# this is needed as a workaround for Tables.jl dispatch
+function DataFrame(pairs::AbstractVector{<:Pair}; makeunique::Bool=false,
+                   copycols::Bool=true)
+    if isempty(pairs)
+        return DataFrame()
+    else
+        if !(all(((k,v),) -> k isa Symbol, pairs) || all(((k,v),) -> k isa AbstractString, pairs))
+            throw(ArgumentError("All column names must be Symbols or strings"))
```

Comment:
```suggestion
            throw(ArgumentError("All column names must be either Symbols or strings"))
```

Reply:
Did you leave out "(mixing is not allowed)" on purpose?

Reply:
No - I just reused error message from above (as it was identical). I now fixed them both. Actually in the tests I added a check against disallowing mixing.